### PR TITLE
Feature search for files

### PR DESCRIPTION
**Pull request by algern0p**
Initiales write up, für locate und find command.

### CODE REVIEW

File: 01_basic_commands/020_search_files/search_files.md [hunk 1]
```diff
@@ -1 +1,139 @@
-Search for files
+# Search For Files
+
+Multiple approaches for searching for files are offered. If you are searching for a file by its file name `locate` is the choice, `find` is used to search for files by its attributes, `grep` allows searching for files via their content.
```
Comment (philippabele) [inline 01_basic_commands/020_search_files/search_files.md:3]: Why are you mentioning grep? I did not find any explanation of the grep in the following sections.

File: 01_basic_commands/020_search_files/search_files.md [hunk 2]
```diff
@@ -1 +1,139 @@
...
+
+## The `locate` command
+
+Locate performs a database search for paths matching the search term.
+
+``` bash
+locate Pictures/Screenshot
+```
+The above command will search for all files and folders starting with `Screenshot-` laying under a location containing `Pictures`. It simply searches for the specified search term in the full path of a file. An exemplary result of the command can be seen below:
+
```
Comment (philippabele) [inline 01_basic_commands/020_search_files/search_files.md:13]: Examples are great, but concrete tasks are better. Think of a are more interactive style of learning thinks. Prepare sample data by which the student has to do something on its own.

File: 01_basic_commands/020_search_files/search_files.md [hunk 3]
```diff
@@ -1 +1,139 @@
...
+```
+/home/user/Desktop/backup/Pictures/Screenshots
+/home/user/Desktop/backup/Pictures/Screenshots/35574917.png
+/home/user/Desktop/backup/Pictures/Screenshots/Screenshot_20210208_093220.png
+/home/user/Pictures/Screenshot_20210921_224523.png
+```
+
+## The `find` command
```
Comment (philippabele) [inline 01_basic_commands/020_search_files/search_files.md:21]: This document is far too long. Think of yourself learning new things. Wouldn't it be great to read something and immediately try it out on your own? Give the reader of your content smaller chunks. Split up this document into smaller parts.

File: 01_basic_commands/020_search_files/search_files.md [hunk 4]
```diff
@@ -1 +1,139 @@
...
+
+``` bash
+find ~ \( -name "*.png" -or -name "*.pdf" \) -type f -user dummy
+```
+
+### Actions
+
+Find can be used to execute actions on the files found. 
+
+#### Predefined Actions
```
Comment (philippabele) [inline 01_basic_commands/020_search_files/search_files.md:92]: You don't have to reinvent the wheel. Give hints to the chapter in "The Linux command line", "--help" pages or the "man pages". Try not the repeat content which is already available there. Only give pointers what the student have to read. Focus yourself on simple practical exercises and a sample solution.

**Review by philippabele — CHANGES_REQUESTED** [review]
Great content, but we have to improve our working style to be more agile.

- split the content into smaller *.md files
- prepare sample practical exercises and explain what to do in theses *.md files
- prepare solutions for your exercises in *.sh files

Most import:

- commit earlier and more often
- I do not believe, that you wrote this document in just one sequence
- PR more often to get more feedback from me
- please let me know, if we have to speak with each other on the phone

**Review by philippabele — CHANGES_REQUESTED** [review]
We are almost there. The split files looks great and I had a lot of fun playing around with your exercises. Still there is some work to do:

- convert all files from CRLF to LF

By running `cat -A 021_solution.sh` it is apparent, that files are made in an Windows environment. Convert them to be useful in an Linux environment. [Hint1](https://stackoverflow.com/questions/14219092/bash-script-and-bin-bashm-bad-interpreter-no-such-file-or-directory) [Hint2](https://unix.stackexchange.com/questions/79702/how-to-test-whether-a-file-uses-crlf-or-lf-without-modifying-it)

Maybe you could add an git attributes to convert all CRLF line endings to LF. That would be the safest solution for further commits and PRs.

**Comment by algern0p** [comment]
> We are almost there. The split files looks great and I had a lot of fun playing around with your exercises. Still there is some work to do:
> 
> * convert all files from CRLF to LF
> 
> By running `cat -A 021_solution.sh` it is apparent, that files are made in an Windows environment. Convert them to be useful in an Linux environment. [Hint1](https://stackoverflow.com/questions/14219092/bash-script-and-bin-bashm-bad-interpreter-no-such-file-or-directory) [Hint2](https://unix.stackexchange.com/questions/79702/how-to-test-whether-a-file-uses-crlf-or-lf-without-modifying-it)
> 
> Maybe you could add an git attributes to convert all CRLF line endings to LF. That would be the safest solution for further commits and PRs.

The files seem to be LF on my end, as they were created on a Linux environment. I cloned the repo into another location to see if git alters them, but they still seem to be LF.

**Comment by philippabele** [comment]
Okay. It seems to be a Windows topic. I use WSL, cloned the git with cmd.exe `git clone ...` and run VS Code with `code .`

I changed my approach to start VS Code from inside WSL and all files work.

**Pull request merged**